spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultPublisherFactory.java

```diff
@@ -56,9 +58,12 @@
 
 	private final CredentialsProvider credentialsProvider;
 
-	public DefaultPublisherFactory(String projectId, ExecutorProvider executorProvider,
-			ChannelProvider channelProvider, CredentialsProvider credentialsProvider) {
-		this.projectId = projectId;
+	public DefaultPublisherFactory(GcpProjectIdProvider projectIdProvider,
+			ExecutorProvider executorProvider,
+			ChannelProvider channelProvider,
+			CredentialsProvider credentialsProvider) {
+		this.projectId = projectIdProvider.getProjectId()
+				.orElseThrow(UnexistingProjectIdException::new);
```

Comment:
The problem with `Optional` that you can't assume that `getProjectId()` won't return `null`...